TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilege.php

```diff
@@ -38,34 +38,53 @@ class CreateNodePrivilege extends AbstractNodePrivilege
      */
     public function matchesSubject(PrivilegeSubjectInterface $subject)
     {
-        if ($subject instanceof CreateNodePrivilegeSubject === false && $subject instanceof MethodPrivilegeSubject === false) {
-            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "%s" only support subjects of type "%s" or "%s", but we got a subject of type: "%s".', CreateNodePrivilege::class, CreateNodePrivilegeSubject::class, MethodPrivilegeSubject::class, get_class($subject)), 1417014353);
+        if (!($subject instanceof CreateNodePrivilegeSubject) && !($subject instanceof MethodPrivilegeSubject)) {
+            throw new InvalidPrivilegeTypeException(
+                sprintf(
+                    'Privileges of type "%s" only support subjects of type "%s" or "%s", but we got a subject of type: "%s".',
+                    CreateNodePrivilege::class,
+                    CreateNodePrivilegeSubject::class,
+                    MethodPrivilegeSubject::class,
+                    \get_class($subject)
+                ),
+                1417014353
+            );
         }
 
         $this->initialize();
+        $allowedCreationNodeTypes = $this->nodeContext->getCreationNodeTypes();
         if ($subject instanceof MethodPrivilegeSubject) {
-            if ($this->methodPrivilege->matchesSubject($subject) === false) {
+            if (!$this->methodPrivilege->matchesSubject($subject)) {
```

Comment:
Again, I'm not sure whether we should change `$condition === false` to `!$condition`

Reply:
I'm no fan of `=== true` or `=== false` if the condition is a bool in any case.
Is there a general code style rule to that? I did not see this explicit type check in many places here where it wasn't needed (in opposition to code with f.e. `strpos()`).

Reply:
In the [CGL](https://flowframework.readthedocs.io/en/stable/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.html#php-in-general) it says:
> Prefer strict comparisons whenever possible, to avoid problems with truthy and falsy values that might behave different than what you expect.

you are right, we don't do that everywhere. especially for cases where we can be sure that the condition is a boolean. But we should have good reason to make a comparision less strict. Not being a fan is not enough I'm afraid :)